### ISSUE TITLE
False positive on Integer Overflow

### ISSUE THREAD

**Issue by havu0** [ConsenSys/mythril#145]
version : v0.16.23

I used mythril to detect vulnerabilities in the sample project
but integer overflow scan detected false positives.

[RBAC.sol]
```
pragma solidity ^0.4.21;

import "./Roles.sol";


/**
 * @title RBAC (Role-Based Access Control)
 * @author Matt Condon (@Shrugs)
 * @dev Stores and provides setters and getters for roles and addresses.
 * @dev Supports unlimited numbers of roles and addresses.
 * @dev See //contracts/mocks/RBACMock.sol for an example of usage.
 * This RBAC method uses strings to key roles. It may be beneficial
 *  for you to write your own implementation of this interface using Enums or similar.
 * It's also recommended that you define constants in the contract, like ROLE_ADMIN below,
 *  to avoid typos.
 */
contract RBAC {
  using Roles for Roles.Role;

  mapping (string => Roles.Role) private roles;

  event RoleAdded(address addr, string roleName);
  event RoleRemoved(address addr, string roleName);

  /**
   * @dev reverts if addr does not have role
   * @param addr address
   * @param roleName the name of the role
   * // reverts
   */
  function checkRole(address addr, string roleName)
    view
    public
  {
    roles[roleName].check(addr);
  }

  /**
   * @dev determine if addr has role
   * @param addr address
   * @param roleName the name of the role
   * @return bool
   */
  function hasRole(address addr, string roleName)
    view
    public
    returns (bool)
  {
    return roles[roleName].has(addr);
  }

  /**
   * @dev add a role to an address
   * @param addr address
   * @param roleName the name of the role
   */
  function addRole(address addr, string roleName)
    internal
  {
    roles[roleName].add(addr);
    emit RoleAdded(addr, roleName);
  }

  /**
   * @dev remove a role from an address
   * @param addr address
   * @param roleName the name of the role
   */
  function removeRole(address addr, string roleName)
    internal
  {
    roles[roleName].remove(addr);
    emit RoleRemoved(addr, roleName);
  }

  /**
   * @dev modifier to scope access to a single role (uses msg.sender as addr)
   * @param roleName the name of the role
   * // reverts
   */
  modifier onlyRole(string roleName)
  {
    checkRole(msg.sender, roleName);
    _;
  }

  /**
   * @dev modifier to scope access to a set of roles (uses msg.sender as addr)
   * @param roleNames the names of the roles to scope access to
   * // reverts
   *
   * @TODO - when solidity supports dynamic arrays as arguments to modifiers, provide this
   *  see: https://github.com/ethereum/solidity/issues/2467
   */
  // modifier onlyRoles(string[] roleNames) {
  //     bool hasAnyRole = false;
  //     for (uint8 i = 0; i < roleNames.length; i++) {
  //         if (hasRole(msg.sender, roleNames[i])) {
  //             hasAnyRole = true;
  //             break;
  //         }
  //     }

  //     require(hasAnyRole);

  //     _;
  // }
}
```
[Roles.sol]
```
pragma solidity ^0.4.21;


/**
 * @title Roles
 * @author Francisco Giordano (@frangio)
 * @dev Library for managing addresses assigned to a Role.
 *      See RBAC.sol for example usage.
 */
library Roles {
  struct Role {
    mapping (address => bool) bearer;
  }

  /**
   * @dev give an address access to this role
   */
  function add(Role storage role, address addr)
    internal
  {
    role.bearer[addr] = true;
  }

  /**
   * @dev remove an address' access to this role
   */
  function remove(Role storage role, address addr)
    internal
  {
    role.bearer[addr] = false;
  }

  /**
   * @dev check if an address has this role
   * // reverts
   */
  function check(Role storage role, address addr)
    view
    internal
  {
    require(has(role, addr));
  }

  /**
   * @dev check if an address has this role
   * @return bool
   */
  function has(Role storage role, address addr)
    view
    internal
    returns (bool)
  {
    return role.bearer[addr];
  }
}
```

```

==== Integer Underflow ====
Type: Warning
Contract: Roles
Function name: hasRole(address,string)
PC address: 555
A possible integer underflow exists in the function hasRole(address,string).
The subtraction may result in a value < 0.
--------------------
In file: /tmp/tmppfpvycmw/contracts/ownership/rbac/Roles.sol:12

 bool) bearer;
  }

  /**
   *

--------------------

==== Integer Underflow ====
Type: Warning
Contract: Roles
Function name: hasRole(address,string)
PC address: 537
A possible integer underflow exists in the function hasRole(address,string).
The subtraction may result in a value < 0.
--------------------
In file: /tmp/tmppfpvycmw/contracts/ownership/rbac/Roles.sol:7

a Role.
 *

--------------------
```

**Comment by JoranHonig** [comment]
@b-mueller This probably has to do with the wrong max UINT value, as I cant reproduce this in the master version of mythril.

**Comment by b-mueller** [comment]
@JoranHonig, I'm getting the same result as above with 0.16.23 and master.

**Comment by JoranHonig** [comment]
So that would indicate that a dependency difference causes this problem. ( 0.16.23 does indeed not include the UINT value)

**Comment by b-mueller** [comment]
Fixed by #146.
